ERCS/erc-7914.md

```diff
@@ -0,0 +1,157 @@
+- `recipient`: The recipient of the Native transfer.
+- `amount`: The amount of Native to transfer.
+
+SHOULD early return when amount == 0, otherwise anyone can execute any fallback function on any contract on behalf of the user without explicit approval.
+
+MUST verify whether `msg.sender` (the caller) is authorized to execute the Native transfer.
+
+MUST handle the Native transfer and return false on failure.
+
+MUST emit the following event on success:
+
+```solidity
+event TransferFromNative(address indexed from, address indexed to, uint256 value);
+```
+
+### Persistent Approval
+
+An approval system similar to the ERC-20 `approve` mechanism MUST be implemented to allow users to specify trusted accounts that can perform Native transfers on their behalf.
+
+To modify this approval, the following function MUST be implemented.
+
+```solidity
+function approveNative(address spender, uint256 amount) external returns (bool);
+```
+
+- `spender`: The address being authorized.
+- `amount`: The maximum Native amount the `spender` can transfer.
+
+This function SHOULD only be callable by the owner of the account.
+
+The following event MUST be emitted when approvals are updated:
+
+```solidity
+event ApproveNative(address indexed owner, address indexed spender, uint256 value);
+```
+
+### Optional Transient Approval
+
+For cases where approval should only persist within a single transaction, a transient approval mechanism MAY be used. This allows a user to grant temporary approval to a `spender` for a single transaction, without modifying persistent storage.
+
+The recommended function for transient approvals is:
+
+```solidity
+function approveNativeTransient(address spender, uint256 amount) external returns (bool);
+```
+
+- `spender`: The address being authorized.
+- `amount`: The Native amount the `spender` is allowed to transfer within the scope of the transaction.
+
+A corresponding event SHOULD be emitted:
+
+```solidity
+event ApproveNativeTransient(address indexed owner, address indexed spender, uint256 value);
+```
+
+### ERC-165 Interface
+
+The integrating contract SHOULD implement the interface as defined by ERC-165 to notify integrating contracts of interface support.
+
+```solidity
+interface ERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165. This function
+    ///  uses less than 30,000 gas.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+```
```

Comment:
imo, to avoid inheritance headaches, would prefer if there's some way to recycle one of the existing functions.

I think if we have some `nativeAllowance(address spender)`, we can use it as a signaling mechanism.

Technically, `transferFromNative` with a zero amount could be used to signal too.